lang_java/tree_sitter/codegen_grammar.py

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/python3
+
+import json
+
+with open("grammar.json", 'r') as fs:
+    data = json.load(fs)
+
+codegen = '''
+(* 
+ * Yoann Padioleau, Sharon Lin
+ * 2020 initial draft
+ *)
+
+(*****************************************************************************)
+(* Prelude *)
+(*****************************************************************************)
+(*
+ * An AST for Java.
+ *
+ * (Not yet complete, this requires further manual edits in order to compile)
+ *
+ *)
+ (*****************************************************************************)
+(* The Tree-sitter AST java related types *)
+(*****************************************************************************)
+(* ------------------------------------------------------------------------- *)
+(* Token/info *)
+(* ------------------------------------------------------------------------- *)
+
+type 'a wrap  = 'a * string
+'''
+
+IGNORE_TYPES = ["PREC_DYNAMIC", "PREC_LEFT", "PREC", "PREC_RIGHT", "SEQ", "STRING", "ALIAS"]
```

Comment:
When do you use this IGNORE_TYPES?

Reply:
I was previously not generating expressions for matching strings and patterns, but that's outdated now, I'll remove.